src/java/org/apache/cassandra/utils/JVMStabilityInspector.java

```diff
@@ -139,7 +139,15 @@ else if (t instanceof UnrecoverableIllegalStateException)
             if (t instanceof FSError || t instanceof CorruptSSTableException)
                 isUnstable = true;
 
-        fn.accept(t);
+        try
+        {
+            fn.accept(t);
```

Comment:
this looked very critical to me; `die` policy stopped working...

Reply:
I looked at how this bug happened and changed the code to match the old semantics as best as I could.  This doesn't solve the problem that the functions can throw; so I keep this logic here to protect against those cases.